core/src/main/java/org/elasticsearch/action/admin/indices/validate/query/TransportValidateQueryAction.java

```diff
@@ -182,11 +182,10 @@ protected ShardValidateQueryResponse shardOperation(ShardValidateQueryRequest re
             searchContext.preProcess();
 
             valid = true;
-            if (request.explain()) {
-                explanation = searchContext.parsedQuery().query().toString();
-            }
             if (request.rewrite()) {
                 explanation = getRewrittenQuery(searcher.searcher(), searchContext.query());
+            } else if (request.explain()) {
+                explanation = searchContext.filteredQuery().query().toString();
```

Comment:
Sorry for moving the code around here. The big change is using `filteredQuery` instead of `parsedQuery`.